feat/feature.go

```diff
@@ -105,6 +105,38 @@ type Collection interface {
 	Location() Feature
 }
 
+// BaseStartOf returns the start of the given feature relative to the deepest
+// location, which may be the feature itself if it has a nil Location.
+// BaseStartOf will panic if the feature chain is deeper than 1000 links.
+func BaseStartOf(f Feature) int {
+	return BasePosOf(f.Location(), f.Start())
+}
+
+// BaseEndOf returns the end of the given feature relative to the deepest
+// location, which may be the feature itself if it has a nil Location.
+// BaseEndOf will panic if the feature chain is deeper than 1000 links.
+func BaseEndOf(f Feature) int {
+	return BasePosOf(f.Location(), f.End())
+}
+
+// BasePosOf converts a position that is defined on a given feature to a
```

Comment:
"BasePosOf returns the position in f converted to coordinates in the deepest feature location and the reference feature. BasePosOf will panic if the feature chain is deeper than 1000 links." and implement this.

This means that position must be checked for being within f (return 0 and nil if not in f).